daemon/stats_collector.go

```diff
@@ -76,22 +76,42 @@ func (s *statsCollector) unsubscribe(c *Container, ch chan interface{}) {
 }
 
 func (s *statsCollector) run() {
+	type publishersPair struct {
+		container *Container
+		publisher *pubsub.Publisher
+	}
+	// we cannot determine the capacity here.
+	// it will grow enough in first iteration
+	pairs := make([]publishersPair, 0)
```

Comment:
This is probably not so important, but I think it is better to specify empty slice as:
```
var pairs []publishersPair
```
I tried - it works okay with slicing to zero.